@xen-orchestra/lite/src/App/Pool/dashboard/ObjectStatus.tsx

```diff
@@ -0,0 +1,166 @@
+import Box from '@mui/material/Box'
+import CircularProgress from '@mui/material/CircularProgress'
+import Grid from '@mui/material/Grid'
+import React from 'react'
+import styled from 'styled-components'
+import Typography from '@mui/material/Typography'
+import { withState } from 'reaclette'
+
+import Icon from '../../../components/Icon'
+import IntlMessage from '../../../components/IntlMessage'
+import { ObjectsByType, Vm, Host } from '../../../libs/xapi'
+
+interface ParentState {
+  objectsByType?: ObjectsByType
+}
+
+interface State {}
+
+interface Props {
+  type: 'host' | 'VM'
+}
+
+interface ParentEffects {}
+
+interface Effects {}
+
+interface Computed {
+  objects?: Map<string, Host> | Map<string, Vm>
+  nTotal?: number
+  nActive?: number
+  nInactive?: number
+}
+
+const DEFAULT_MARGIN = { mt: 2 }
+const TYPOGRAPHY_SX = { mb: 2 }
+
+const ObjectStatusContainer = styled.div`
+  display: flex;
+  overflow: hidden;
+  flex-direction: row;
+  align-content: space-between;
+  margin-bottom: 1em;
+`
+
+const CircularProgressPanel = styled.div`
+  margin-left: 2em;
+`
+
+const GridPanel = styled.div`
+  margin-left: 2em;
+  width: 100%;
+  height: 100%;
+`
+
+const getHostPowerState = (host: Host) => {
+  const { $metrics } = host
+  return $metrics ? ($metrics.live ? 'Running' : 'Halted') : 'Unknown'
+}
+
+// TODO: use CircularProgress component when https://github.com/vatesfr/xen-orchestra/pull/6128 is merged
+const ObjectStatus = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    computed: {
+      objects: (state, { type }) =>
+        type === 'VM'
+          ? state.objectsByType
+              ?.get(type)
+              ?.filter((vm: Vm) => !vm.is_control_domain && !vm.is_a_snapshot && !vm.is_a_template)
+          : state.objectsByType?.get(type),
+      nTotal: ({ objects }) => objects?.size,
+      nActive: ({ objects }, { type }) =>
+        (type === 'VM'
+          ? objects?.filter((vm: Vm) => vm.power_state === 'Running')
+          : objects?.filter((host: Host) => getHostPowerState(host) === 'Running')
+        ).size,
+      nInactive: ({ nTotal = 0, nActive = 0 }) => nTotal - nActive,
+    },
+  },
+  ({ state: { nTotal, nActive = 0, nInactive }, type }) => {
+    if (nTotal === undefined) {
+      return (
+        <span>
+          <IntlMessage id={type === 'VM' ? 'noVms' : 'noHosts'} />
+        </span>
+      )
+    }
+
+    return (
+      <ObjectStatusContainer>
+        <CircularProgressPanel>
+          <Box sx={{ position: 'relative', display: 'inline-flex' }}>
+            <CircularProgress
+              variant='determinate'
+              value={(nActive * 100) / nTotal}
```

Comment:
May `nTotal` be `undefined`?

Reply:
And `nActive` may also be `undefined`.

Reply:
Also, in the mockup, when data is not loaded yet, it shows a spinner. Should it be in another PR?